pytket/pytket/wasm/wasm.py

```diff
@@ -153,51 +159,69 @@ def __init__(self, filepath: str, check_file: bool = True, int_size: int = 32):
                 for t in function_signatures[self._function_types[idx]]["return_types"]:
                     if t != self._int_type:
                         supported_function = False
-            else:
-                supported_function = False
-
-            if supported_function:
-                self._functions[x] = (
-                    len(
-                        function_signatures[self._function_types[idx]][
-                            "parameter_types"
-                        ]
-                    ),
-                    len(function_signatures[self._function_types[idx]]["return_types"]),
-                )
 
-            if not supported_function:
-                self._unsupported_function.append(x)
+                if supported_function:
+                    self._functions[x] = (
+                        len(
+                            function_signatures[self._function_types[idx]][
+                                "parameter_types"
+                            ]
+                        ),
+                        len(
+                            function_signatures[self._function_types[idx]][
+                                "return_types"
+                            ]
+                        ),
+                    )
+
+                if not supported_function:
+                    self._unsupported_function.append(x)
+
+        except:
```

Comment:
I'm wary of catching all errors, at least if we swallow up the error content like this -- it can easily mask bugs. Why don't we define a custom error type that we can throw from specific places in this code? Are there particular cases you're aware of where the checker fails but the code runs? Ideally this would not be the case but if there are I think it would be better to single them out rather than treat all errors this way.

Reply:
The general idea if this was to avoid cryptic error messages when paring wasm files that are broken. I wanted to point out to the user that something has gone wrong and explain what the causes might have been. I am not aware of wasm files that can break the parsing and still can be run. I have removed this now.

Reply:
Removed in [71697f1](https://github.com/CQCL/tket/pull/1070/commits/71697f1aa1ba6074499177144e71b8c50c85b037)